mlem/core/data_type.py

```diff
@@ -577,6 +613,135 @@ def read_batch(
         raise NotImplementedError
 
 
+class DynamicDictType(DataType, DataSerializer):
+    """
+    Dynamic DataType for dict without fixed set of keys
+    """
+
+    type: ClassVar[str] = "d_dict"
+
+    key_type: DataType
```

Comment:
keys should be immutable. and json keys are always strings. Since you write it as json, how would you recreate arbitrary type?